### DISCUSSION TITLE
Magento Skeleton

### DISCUSSION THREAD

**Discussion by anthony-blackbird** [php/frankenphp#640] [Show and tell]
Hello Kevin,
We are working with french community to setup Magento Skeleton for FrankenPHP here : https://github.com/opengento/magento2-frankengento.

We would appreciate it if you can share it on https://frankenphp.dev/fr/docs/#exemples-et-squelettes page for the community
We have quite good performance in our first tests (TTFB divided by 2 !), great job !

**Comment by dunglas** [comment]
Thank you for this Magento skeleton!!
We'll add it to the docs and website.

**Comment by ryansunxl** [comment]
Bump this up. @anthony-blackbird 's repo is the only one I found that supports worker mode, which I think is the biggest advantage of frankenphp. I would suggest promoting this one over others. 

However, while testing, I don't find dramatic improvement on TTFB with worker mode, perhaps it is because magento is bootstrapped inside request handler? 
https://github.com/opengento/magento2-frankengento/blob/2c4a6c0755d546db3c9fc818aa16ebf36e53183a/app/pub/index.php.sample#L33-L36

**Reply by anthony-blackbird** [reply]
To fully leverage FrankenPHP with Magento, the ObjectManager would need to be shared between processes. However, in Magento's current bootstrap architecture, the ObjectManager is instantiated within each request and is not shared. One potential approach to benefit from FrankenPHP’s persistent model could be to create a dedicated bootstrap layer for specific shared services or lightweight objects—such as factories—that could be safely reused across PHP stacks.

This would require careful design to ensure compatibility and thread safety, but it might offer meaningful performance improvements in certain scenarios.

In this current index.php.sample the only shared opperation is actually: 
```
require __DIR__ . '/../app/bootstrap.php';
```
